docker/dockerfiles/base

```diff
@@ -15,21 +18,17 @@ RUN apt-get update && apt-get install apt-transport-https && \
             gettext build-essential \
             libxml2-dev libxslt1-dev zlib1g-dev git \
             libjpeg-dev libffi-dev libssl-dev libxslt1.1 \
-            libmariadbclient-dev mariadb-client optipng \
-            nodejs=0.10.48-1nodesource1~jessie1 && \
+            libmariadbclient-dev mariadb-client && \
     rm -rf /var/lib/apt/lists/*
 
-COPY ./requirements /app/requirements
-
-RUN pip install --no-cache-dir -r requirements/default.txt --require-hashes
-RUN pip install --no-cache-dir -r requirements/dev.txt --require-hashes
-RUN pip install --no-cache-dir -r requirements/test.txt --require-hashes
-
-COPY ./package.json /app/package.json
-COPY ./bower.json /app/bower.json
+COPY ./requirements/ /app/requirements/
 
-RUN npm install
+RUN bash -c '\
+    virtualenv /venv && \
```

Comment:
Why do we need a virtualenv? I've never run into issues when using the system python in the python images.

Reply:
I see. You're copying the venv to other images and this one is not inherited from for the prod image. Interesting.

Reply:
I used a virtualenv to contain everything so I can easily copy them other to the final image. The final docker image is based on top of python-slim and does not include all the compiler / dev libs required to install the python packages. 

This reduces attack surface and limits the size of the final image. win-win!

Reply:
Great point! Didn't think of that. I'll look at that technique for bedrock as well. 🎉